ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextShadowNode.java

```diff
@@ -439,8 +452,18 @@ public void setTextAlign(@Nullable String textAlign) {
 
   @ReactProp(name = ViewProps.FONT_SIZE, defaultFloat = UNSET)
   public void setFontSize(float fontSize) {
-    if (fontSize != UNSET) {
-      fontSize = (float) Math.ceil(PixelUtil.toPixelFromSP(fontSize));
+    setFontSize(fontSize, false);
+  }
+
+  public void setFontSize(float fontSize, boolean forceRecalculation) {
```

Comment:
Essentially I don't want to call setFontSize with an `UNSET` value.  If I do so there are issues with the font displaying correctly.  I took your suggestion and removed `forceRecalculation` but instead have to check for `fontSize == UNSET` and if so set it to `ViewDefaults.FONT_SIZE_SP` so that fontSize is set to 14 (current default) rather than -1.

While updating it I also have `setAllowFontScaling` only call `setFontSize` if the value of `allowFontScaling` actually changed.